### DISCUSSION TITLE
LVN encountering non-local names (Lesson 3)

### DISCUSSION THREAD

**Discussion by artjomPlaunov** [sampsyo/cs6120#261] [Q&A]
As the name implies, local value numbering does not deal with control flow between blocks, so in the case where a non local variable is used as an argument, it is left as is. 

My question is, can we still run the algorithm on the rest of the block, or should the numbering of the block be short circuited and we simply return the block as it was?

**Comment by sampsyo** [comment]
I think an example could help illustrate the options you have in mind!

But in case it helps, I definitely don't think a single "external input" variable should ruin LVN for the rest of the block. It should be possible to optimize subsequent instructions, especially if they do not use the input variable in question.
